app/src/main/java/ua/kozlov/madfood/Foods.java

```diff
@@ -0,0 +1,265 @@
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class Foods implements FoodsInterface {
+    @Override
+    public ArrayList<String> getFoodsList(String groupName, Context context) {
+        XmlPullParser parser = getParser(groupName, context);
+        ArrayList<String> foodsList = setFoodNamesToList(parser);
+        return foodsList;
+    }
+
+    @Override
+    public String getFoodName(String groupID, String id, Context context) {
+        return null;
+    }
+
+    @Override
+    public String getFoodId(String groupID, String name, Context context) {
+        return null;
+    }
+
+    @Override
+    public ArrayList<String> getFoodParameters(String groupName, int position, Context context) {
+        XmlPullParser parser = getParserFacts(groupName, context);
+        ArrayList<String> foodsList = setFoodParametersToList(parser, position);
+        return foodsList;
+    }
+
+    private ArrayList<String> setFoodNamesToList(XmlPullParser parser) {
+        ArrayList<String> list = new ArrayList<>();
+        try {
+            while (parser.getEventType() != XmlPullParser.END_DOCUMENT) {
+                if (parser.getEventType() == XmlPullParser.START_TAG
+                        && parser.getName().equals("Food")) {
+                    final String foodName = parser.getAttributeValue(1);
+                    list.add(foodName);
+                }
+                parser.next();
+            }
+        } catch (XmlPullParserException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return list;
+    }
+
+    private ArrayList<String> setFoodParametersToList(XmlPullParser parser, int position) {
+        ArrayList<String> list = new ArrayList<>();
+        int count = 0;
+        try {
+            while (parser.getEventType() != XmlPullParser.END_DOCUMENT) {
+                if (parser.getEventType() == XmlPullParser.START_TAG
+                        && parser.getName().equals("Food")) {
+                    if (count == position){
+                        list.add(parser.getAttributeValue(1));
+                        list.add(parser.getAttributeValue(2));
+                        list.add(parser.getAttributeValue(3));
+                        list.add(parser.getAttributeValue(4));
+                        list.add(parser.getAttributeValue(5));
+                        break;
+                    } else {
+                        count++;
+                    }
+                }
+                parser.next();
+            }
+        } catch (XmlPullParserException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return list;
+    }
+
+    private XmlPullParser getParser(String group, Context context) {
+        XmlPullParser parser = null;
+        if (group.equals("Vegetables")) {
```

Comment:
@OlehKozlov  this method looks ugly, spaghetti code
can you modify it, maybe with **switch** constuction?